### DISCUSSION TITLE
how to handle unit test error multiple element

### DISCUSSION THREAD

**Discussion by shipcake** [vercel/next.js#71300] [Help]
### Summary

Hi I have a probleam with vitest library. I beginner of use unit testing but I found a probleam from vitest and @testing-library/react

![image](https://github.com/user-attachments/assets/51f57a9b-4476-4c66-81cf-c0855d59c8d9)

I sure it have one element but I not sure why. from now I tempolary fix with use get**All**ByTestId

My example code testing file
`it("should display validation errors for invalid input",async () => {
  render(<MyForm />)
  const user = userEvent.setup()
  await user.click(screen.getByRole("button", { name: /submit/i }))

  expect(screen.getByTestId("name_err_msg")).toBe("Name must be at least 2 characters");
  expect(screen.getByTestId("email_err_msg")).toBe("Invalid email address");
})`

**Comment by wilsonadenuga** [comment]
Can you paste what the form looks like and your test too?

**Reply by shipcake** [reply]
yes 
`
<form onSubmit={handleSubmit(onSubmit)}>
      <div>

        <label htmlFor="name">Name:</label>
        <input type="text" data-testid="testname" {...register("name")} />
        {errors.name && <span role="alert" data-testid="name_err_msg">{errors.name.message}</span>}
      </div>
      <div>
        <label htmlFor="email">Email:</label>
        <input type="email" id="email" data-testid="email"  {...register("email")} />
        {errors.email && <span role="alert" data-testid="email_err_msg">{errors.email.message}</span>}
      </div>
      <button type="submit" data-testid="submitbtn" className="bg-teal-400">Submit</button>
    </form>
`
